jobrunner/controller/main.py

```diff
@@ -248,6 +248,22 @@ def handle_job(job, mode=None, paused=None):
                 results = JobTaskResults.from_dict(task.agent_results)
                 save_results(job, results)
                 # TODO: Delete obsolete files
+        elif task.agent_stage is not None and task.agent_stage != job.status_code:
```

Comment:
Do we need an extra branch here? What would happen if we ran `set_code(job, StatusCode.from_value ...)` for every not yet completed job and got rid of `refresh_job_timestamps()` altogether?

Reply:
good point, will update